satellite/payments/coinpayments/transactions.go

```diff
@@ -0,0 +1,320 @@
+	cmdGetTransactionInfo     = "get_tx_info"
+	cmdGetTransactionInfoList = "get_tx_info_multi"
+)
+
+// Currency is a type wrapper for defined currencies.
+type Currency string
+
+const (
+	// CurrencyUSD defines USD.
+	CurrencyUSD Currency = "USD"
+	// CurrencyLTCT defines LTCT, coins used for testing purpose.
+	CurrencyLTCT Currency = "LTCT"
+)
+
+// String returns Currency string representation
+func (c Currency) String() string {
+	return string(c)
+}
+
+// Status is a type wrapper for transaction statuses.
+type Status int
+
+const (
+	// StatusCancelled defines cancelled or timeout transaction.
+	StatusCancelled Status = -1
+	// StatusPending defines pending transaction which is waiting for buyer funds.
+	StatusPending Status = 0
+	// StatusReceived defines transaction which successfully received required amount of funds.
+	StatusReceived Status = 1
+)
+
+// Int returns int representation of status.
+func (s Status) Int() int {
+	return int(s)
+}
+
+// String returns string representation of status.
+func (s Status) String() string {
+	switch s {
+	case StatusCancelled:
+		return "cancelled/timeout"
+	case StatusPending:
+		return "pending"
+	case StatusReceived:
+		return "received"
+	default:
+		return "unknown"
+	}
+}
+
+// TransactionID is type wrapper for transaction id.
+type TransactionID string
+
+// String returns string representation of transaction id.
+func (id TransactionID) String() string {
+	return string(id)
+}
+
+// TransactionIDList is a type wrapper for list of transactions.
+type TransactionIDList []TransactionID
+
+// Encode returns encoded string representation of transaction id list.
+func (list TransactionIDList) Encode() string {
+	if len(list) == 0 {
+		return ""
+	}
+	if len(list) == 1 {
+		return string(list[0])
+	}
+
+	var builder strings.Builder
+	for _, id := range list[:len(list)-1] {
+		builder.WriteString(string(id))
+		builder.WriteString("|")
+	}
+
+	builder.WriteString(string(list[len(list)-1]))
+	return builder.String()
+}
+
+// Transaction contains data returned on transaction creation.
+type Transaction struct {
+	ID             TransactionID
+	Address        string
+	Amount         float64
+	DestTag        string
+	ConfirmsNeeded int
+	Timeout        time.Duration
+	CheckoutURL    string
+	StatusURL      string
+	QRCodeURL      string
+}
+
+// UnmarshalJSON handles json unmarshaling for transaction.
+func (tx *Transaction) UnmarshalJSON(b []byte) error {
+	var txRaw struct {
+		Amount         string `json:"amount"`
+		Address        string `json:"address"`
+		DestTag        string `json:"dest_tag"`
+		TxID           string `json:"txn_id"`
+		ConfirmsNeeded string `json:"confirms_needed"`
+		Timeout        int    `json:"timeout"`
+		CheckoutURL    string `json:"checkout_url"`
+		StatusURL      string `json:"status_url"`
+		QRCodeURL      string `json:"qrcode_url"`
+	}
+
+	if err := json.Unmarshal(b, &txRaw); err != nil {
+		return err
+	}
+
+	amount, err := strconv.ParseFloat(txRaw.Amount, 64)
```

Comment:
It feels weird that this is represented as a float64. Are you sure it can handle all the necessary numbers?

e.g. `1.000000000000000000000001`

https://play.golang.org/p/1C_UriE642n

Reply:
Yeah, good point